update SSO integration instructions (Microsoft Entra ID) (#1437)
semgrep/semgrep-docs: docs/deployment/sso.md

1. Sign in to the [Microsoft Entra admin center](https://entra.microsoft.com/).
2. Use the search bar to find and navigate to **enterprise applications**.
   ![Microsoft Entra admin center's Enterprise applications screen](/img/entra-1.png#md-width)
3. Click **New application** > **Create your own application**. A menu appears.
   ![Create your own application screen](/img/entra-2.png#md-width)
4. Name your new application something like `Semgrep SAML`.
5. Select **Integrate any other application you don't find in the gallery (non-gallery)**.
6. Click **Create**. This takes you to your new enterprise application's page.

You have now created a custom enterprise app for Semgrep to integrate with Microsoft Entra ID. This enables you to set up SAML SSO.

#### Set up SAML SSO for your new enterprise app

1. From your new enterprise app's page, go to **Single-sign on** > **SAML**. 
   ![Enterprise application's Single-sign on menu option](/img/entra-3.png#md-width)
2. When prompted to **Select a single sign-on method**, select **SAML**. You are redirected to the **SAML-based Sign-on** page.
   ![SAML-based Sign-on screen](/img/entra-4.png#md-width)

> 3. Click **Edit** to update your **Basic SAML Configuration** by providing the **Entity ID** and **Reply URL**. You can retrieve these values from Semgrep Cloud Platform by performing the following steps:

Reviewer: ```suggestion
3. In the **Basic SAML Configuration** section, click **Edit**. Provide the **Entity ID** and **Reply URL**. You can retrieve these values from Semgrep Cloud Platform by performing the following steps:
```

    1. Log in to Semgrep Cloud Platform and navigate to [**Settings** > **Access** > **SSO**](https://semgrep.dev/orgs/-/settings/access/sso) page. 
    2. Click **Add SAML2 SSO**.
    3. Copy the **Audience URL** value from Semgrep Cloud Platform. Return to **Basic SAML Configuration**. Click **Add identifier** to paste this value as the **Identifier (Entity ID)**.
    4. Copy the **SSO URL** value from Semgrep Cloud Platform. Return to **Basic SAML Configuration**. Click **Add reply URL** to paste this value as the **Reply URL (Assertion Consumer Service URL)**.
4. Click **Save** and close out of **Basic SAML Configuration**.

> 5. Click **Edit** to modify **Attributes & Claims**. You must add two claims. To add your first claim:

Reviewer: ```suggestion
5. In the **Attributes and Claims** section, click **Edit**. You must add two claims. To add your first claim:
```

    1. Click **Add new claim**.
    2. Enter `name` in the **Name** field.

>     3. For the **Source attribute** drop-down box, select `user.givenname`.

Reviewer 2: I know this is what we had before, but I wonder if we should actually be using `user.displayName`? The given name is just the first name which in many orgs is not enough to properly identify a user.

    4. Click **Save**.
6. To add your second claim:
    1. Click **Add new claim**.
    2. Enter `email` in the **Name** field.
    3. From the **Source attribute** drop-down box, select `user.mail`.
    4. Click **Save**.
7. Close out of **Attributes & Claims**.
8. Navigate to Semgrep Cloud Platform, and provide the values required by the SAML2 form:
    1. Provide the **Display name** and the **Email domain** you are using for the integration.
    2. Copy the **Login URL** value from Microsoft Entra ID and paste it in into SCP's **IDP SSO URL** field.
    3. Copy and paste the **Microsoft Entra ID Identifier** value into SCP's **IdP Issuer ID** field.

>     4. In Entra ID's **SAML-based Sing-on** page, click **Download** to obtain the **Certificate (Base64)**.

Reviewer: ```suggestion
    4. In Entra ID's **SAML-based Sign-on** page, click **Download** to obtain the **Certificate (Base64)**.
```
